Automattic/wp-calypso
# Update/remove hardcoded constants preview upgrade nudge

This PR removes usage of `PLAN_*` constants from `preview-upgrade-nudge`

Since we are adding new `2_YEAR` plan constants (p9jf6J-eR-p2), this is a good opportunity to refactor relevant places such as this one instead of just adding another constant to every if.

Test plan:
* Run unit tests
* Go to pages and click "preview" on any page
* In preview window, choose search&social like this:
<img width="611" alt="zrzut ekranu 2018-03-29 o 11 47 27" src="https://user-images.githubusercontent.com/205419/38082682-b2dbb1e2-3347-11e8-8dd6-87c102f1f730.png">

* Confirm that upsell banner displayed for non-business WP.com users looks like this:
<img width="936" alt="zrzut ekranu 2018-03-29 o 11 53 40" src="https://user-images.githubusercontent.com/205419/38082720-d547197e-3347-11e8-97cf-8a46abee1146.png">

* It seems to be non-testable for jetpack sites as preview is not displayed in an overlay, but links to a target site.
